### PR TITLE
Add Solaris targets

### LINKED ISSUE #134: Solaris target
In rust-lang/rust#45001 I was pointed here. I am thinking of adding support for Solaris. @malbarbo asked for SPARC support, but at least a sparc64-unknown-linux-gnu already exists. I would build sparcv9-sun-solaris and x86_64-sun-solaris.

As I read through the sources, they seem to be similar to the Docker CI stuff in rust-lang/rust. Also there is alexcrichton/port-of-rust, which also seems to have the same roots. Would it be preferable to have a [prebuilt](https://github.com/alexcrichton/port-of-rust/tree/master/prebuilt) Solaris Docker image to reuse here?

### PR DESCRIPTION
**Pull request by bgermann**
Closes #134.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -41,10 +41,14 @@ matrix:
...
     # BSD
     - env: TARGET=i686-unknown-freebsd                  DYLIB=1 STD=1 OPENSSL=0.7.17
-    - env: TARGET=x86_64-unknown-dragonfly               DYLIB=1       OPENSSL=0.5.5
+    - env: TARGET=x86_64-unknown-dragonfly              DYLIB=1       OPENSSL=0.5.5
     - env: TARGET=x86_64-unknown-freebsd                DYLIB=1 STD=1 OPENSSL=0.5.5
     - env: TARGET=x86_64-unknown-netbsd                 DYLIB=1 STD=1 OPENSSL=0.7.17
 
+    # Solaris
+    - env: TARGET=sparcv9-unknown-freebsd               DYLIB=1 STD=1 OPENSSL=0.7.17
+    - env: TARGET=x86_64-unknown-freebsd                DYLIB=1 STD=1 OPENSSL=0.7.17
```
Comment (malbarbo) [inline .travis.yml]: Copy and paste typos

**Comment by bgermann** [comment]
I intend to use Solaris 11 cross compiler here, because that is what is used by most people. I actually only use a hack on the rust repo to have a S10 backward compatible stdlib produced. But it also works for S11 targets and is produced with an illumos sysroot. I think I am the only one using Rust on S10 up to now as I had to change some things to make it work some weeks ago.

**Comment by japaric** [comment]
Thanks @bgermann.

I'll prefer to match the rust repo version (i.e. S10) but if it's not that commonly used then S11 sounds fine. This LGTM.

@homunkulus r+

**Comment by homunkulus** [comment]
:pushpin: Commit 02c7b81 has been approved by `japaric`

**Comment by homunkulus** [comment]
:hourglass: Testing commit 02c7b81f0928c374842c41e76efdbe0ce9bfec9b with merge 359c3bb35bcaf47a7193d3ccb2cccbba097da791...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/japaric/cross/builds/290413634?utm_source=github_status&utm_medium=notification)
Approved by: japaric
Pushing 359c3bb35bcaf47a7193d3ccb2cccbba097da791 to master...

**Pull request merged**